Index: app/controllers/wiki_controller.rb
===================================================================
--- app/controllers/wiki_controller.rb	(revision 10697)
+++ app/controllers/wiki_controller.rb	(working copy)
@@ -71,15 +71,22 @@
       return
     end
     @content = @page.content_for_version(params[:version])
+    call_hook(:controller_wiki_decode_content, { :params => params, :content => @content})
     if User.current.allowed_to?(:export_wiki_pages, @project)
       if params[:format] == 'pdf'
+	params[:decode]='1'
+	 call_hook(:controller_wiki_decode_content_export, { :params => params, :content => @content})
         send_data(wiki_page_to_pdf(@page, @project), :type => 'application/pdf', :filename => "#{@page.title}.pdf")
         return
       elsif params[:format] == 'html'
+	params[:decode]='1'
+	 call_hook(:controller_wiki_decode_content_export, { :params => params, :content => @content})
         export = render_to_string :action => 'export', :layout => false
         send_data(export, :type => 'text/html', :filename => "#{@page.title}.html")
         return
       elsif params[:format] == 'txt'
+	params[:decode]='1'
+	 call_hook(:controller_wiki_decode_content_export, { :params => params, :content => @content})
         send_data(@content.text, :type => 'text/plain', :filename => "#{@page.title}.txt")
         return
       end
@@ -109,7 +116,9 @@
 
     # To prevent StaleObjectError exception when reverting to a previous version
     @content.version = @page.content.version
-    
+    params[:decode]='1'
+    call_hook(:controller_wiki_decode_content_with_tags, { :params => params, :content => @content})
+
     @text = @content.text
     if params[:section].present? && Redmine::WikiFormatting.supports_section_edit?
       @section = params[:section].to_i
@@ -149,6 +158,7 @@
       @content.text = @text
     end
     @content.author = User.current
+    call_hook(:controller_wiki_edit_before_save, {:content => @content})
     @page.content = @content
     if @page.save
       attachments = Attachment.attach_files(@page, params[:attachments])
@@ -181,7 +191,7 @@
     @page.update_attribute :protected, params[:protected]
     redirect_to :action => 'show', :project_id => @project, :id => @page.title
   end
-
+   
   # show page history
   def history
     @version_count = @page.content.versions.count
Index: app/views/wiki/show.html.erb
===================================================================
--- app/views/wiki/show.html.erb	(revision 10697)
+++ app/views/wiki/show.html.erb	(working copy)
@@ -9,6 +9,7 @@
 <%= link_to_if_authorized(l(:button_rollback), {:action => 'edit', :id => @page.title, :version => @content.version }, :class => 'icon icon-cancel') unless @content.current_version? %>
 <% end %>
 <%= link_to_if_authorized(l(:label_history), {:action => 'history', :id => @page.title}, :class => 'icon icon-history') %>
+<%= call_hook(:view_wiki_contextual,{:id => @project.identifier, :toggle => params[:decode], :content=>@content}) %>
 </div>
 
 <%= wiki_page_breadcrumb(@page) %>
